main_test.go

```diff
@@ -0,0 +1,41 @@
+package main_test
+
+import (
+	"bytes"
+	"os"
+	"os/exec"
+	"testing"
+	"time"
+
+	"github.com/onsi/gomega"
+	"github.com/onsi/gomega/gexec"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestMainCLI(t *testing.T) {
+	gomega.RegisterTestingT(t)
+	pathToMainCLI, err := gexec.Build("main.go")
```

Comment:
Never worked with gexec but I like it. I was thinking about adding e.g. BATS to the project, but this works a lot better.